cmake/script/GenerateHeaderFromJson.cmake

```diff
@@ -2,25 +2,21 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or https://opensource.org/license/mit/.
 
+get_filename_component(json_source_basename ${JSON_SOURCE_PATH} NAME_WE)
+
 file(READ ${JSON_SOURCE_PATH} hex_content HEX)
-string(REGEX MATCHALL "([A-Za-z0-9][A-Za-z0-9])" bytes "${hex_content}")
+string(REGEX REPLACE "................" "\\0\n" formatted_bytes "${hex_content}")
```

Comment:
nit: Using the `string(REPEAT ...)` command to create a string of 16 dots can improve the code's clarity, I think.

Reply:
I thought of that but couldn't find a way to do it inline - and defing a new variable in a separate line just for this makes it less readable
```cmake
string(REPEAT "." 16 line_width)
string(REGEX REPLACE "${line_width}" "\\0\n" formatted_bytes "${hex_content}")
```